docs(linter): Update various rules to note that they can be disabled for TS code.

connorshea opened the pull request:
This is generally based on the `handled_by_typescript` flag in the original ESLint rule definitions. All of these are flagged in the original rules as being "handled by TypeScript", and so I deferred to that when determining which rules to note this on.

camc314 commented:
### Merge activity

* **Dec 14, 1:37 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 14, 1:37 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Dec 14, 1:42 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Update various rules to note that they can be disabled for TS code. (#16819)

This is generally based on the `handled_by_typescript` flag in the original ESLint rule definitions. All of these are flagged in the original rules as being "handled by TypeScript", and so I deferred to that when determining which rules to note this on.
